File: plugins/de.ovgu.featureide.examples/featureide_examples/Email-FH-AspectJ/features/Sign/EmailSystem/Email.java
package EmailSystem;

public class Email {
	protected boolean signed;
	protected int signkey;
	
	static void printMail(Email msg) {
		original(msg);
		Util.prompt("SIGNED " + msg.isSigned());
		Util.prompt("SIGNATURE " + msg.getEmailSignKey());
	}
	
	void setEmailIsSigned(boolean value) {
		signed = value;
	}

	void setEmailSignKey(int value) {
		signkey = value;
	}	
	
	boolean isSigned() {
		return signed;
	}
	
	int getEmailSignKey() {
		return signkey;
	}
}

File: plugins/de.ovgu.featureide.examples/featureide_examples/EmailSystem-FH-C/features/Sign/Email.c
void
printMail (struct email *msg)
{
  original (msg);
  printf ("SIGNED\n  %i\n", msg->isSigned);
  printf ("SIGNATURE\n  %s\n", msg->signKey);
}

struct email *
cloneEmail (struct email *msg)
{
  struct email *clone = original (msg);
  clone->isSigned = msg->isSigned;
  if (msg->signKey)
    clone->signKey = strdup (msg->signKey);
  return clone;
}

int
isSigned (struct email *msg)
{
  return msg->isSigned;
}
